src/or/connection.c

```diff
@@ -1443,6 +1477,68 @@ connection_listener_new(const struct sockaddr *listensockaddr,
   return NULL;
 }
 
+/**
+ * Create a new listener connection for a given <b>port</b>. In case we
+ * cannot create listener now, but will be able later, set <b>defer</b>
+ * to true. If we cannot bind listening socket because address is already
```

Comment:
I'm not sure that this explanation of defer makes sense.
We unconditionally set defer whenever NoListen is set.
It has nothing to do with whether we can create a listener now, or whether we will be able to later.